tests/load/pipeline/test_pipelines.py

```diff
@@ -997,3 +997,126 @@ def events_timezone_unset():
+    destination_config: DestinationTestConfiguration,
+) -> None:
+    """
+    Test incremental lag behavior when updating previously loaded data.
+
+    Three resources (`r1`, `r2`, `r3`) share the primary key 'id'.
+    - `r1` loads `id=0` and `id=1`.
+    - `r2` adds `id=2` and `id=3`.
+    - `r3`, with lag=1, skips `id=1` and updates entries with `id=2`, `id=3`,
+      and adds a new entry `id=4`. The merge operation reflects these changes.
+
+    We validate that the final dataset correctly orders the updated and new entries,
+    excluding `id=1` from the updates due to the lag.
+    """
+
+    dataset_name = f"{destination_config.destination_name}{uniq_id()}"
+    pipeline = destination_config.setup_pipeline("pipeline", dataset_name=dataset_name)
+
+    name = "items"
+
+    @dlt.resource(name=name, primary_key="id")
+    def r1(_=dlt.sources.incremental("id")):
+        yield from [{"id": 0, "name": "bobby"}, {"id": 1, "name": "jeremy"}]
+
+    @dlt.resource(name=name, primary_key="id")
+    def r2(_=dlt.sources.incremental("id")):
+        yield from [{"id": 2, "name": "james"}, {"id": 3, "name": "john"}]
+
+    # r3 has lag=1, so id=1 is ignored, only id>=2 is updated
+    updated_items = [
+        {"id": 1, "name": "maria"},
+        {"id": 2, "name": "mark"},
+        {"id": 3, "name": "pablo"},
+    ]
+
+    @dlt.resource(name=name, primary_key="id", write_disposition="merge")
+    def r3(_=dlt.sources.incremental("id", lag=1)):
+        yield from [{"id": 4, "name": "max"}] + updated_items
+
+    pipeline.run(r1)
+    pipeline.run(r2)
+    pipeline.run(r3)
+
+    # Validate final dataset
+    with pipeline.sql_client() as sql_client:
+        assert [
+            row[0] for row in sql_client.execute_sql(f"SELECT name FROM {name} ORDER BY id")
+        ] == ["bobby", "jeremy", "mark", "pablo", "max"]
+
+
+@pytest.mark.parametrize(
+    "destination_config",
+    destinations_configs(
+        default_staging_configs=True, default_sql_configs=True, subset=["duckdb", "postgres"]
+    ),
+    ids=lambda x: x.name,
+)
+@pytest.mark.parametrize("lag", [3602, 3601, 3600, 3599, -1])
+def test_pipeline_resource_incremental_datetime_lag(
+    destination_config: DestinationTestConfiguration, lag: float
+) -> None:
+    """
+    Test incremental lag behavior for datetime data while using `id` as the primary key.
+
+    - `r1` loads entries with `id` and `created_at` timestamps.
+    - `r2`, with different lag values, updates entries but only considers records
+      where `created_at` is within the lag window.
+
+    We validate that the final dataset reflects the correct data updates, taking the lag into account.
+    """
+
+    dataset_name = f"{destination_config.destination_name}{uniq_id()}"
+    pipeline = destination_config.setup_pipeline("pipeline", dataset_name=dataset_name)
+
+    name = "events"
+
+    @dlt.resource(name=name, primary_key="id")
+    def r1(_=dlt.sources.incremental("created_at")):
```

Comment:
the concept of test is good. but IMO you should create just one resource with merge write disposition that you call with the lag you want from the very  beginning. this is how people will use it IMO.

emulate returning "updated_events" on the second call to it ie. via some kind of nonlocal flag that tells to add those events on the second time.

the results of the test look good but please test two additional things
- that you do not apply lag to "initial_value" (set it in incremental)
- there's IMO issue with internal deduplication. please test for updated_events that update id=3:
```
{
                "id": 3,
                "created_at": "2023-03-03T02:00:01Z",
                "event": "updated",
            }
```
```
IMO it will not be included into final result because we'll deduplicate it. you should IMO disable deduplication when lag is defined
```py
@property
    def deduplication_disabled(self) -> bool:
        """Skip deduplication when length of the key is 0"""
        return isinstance(self.primary_key, (list, tuple)) and len(self.primary_key) == 0
```